### DISCUSSION TITLE
Passing inputs from frontend to backend

### DISCUSSION THREAD

**Discussion by ambarish999** [tauri-apps/tauri#5141] [Q&A]
Hi,
I've a login app in react js and flask. I was trying to build a desktop version just for fun. I've setup the react js files in the src folder, and the flask as a sidecar. My question is: **How do I pass the inputs from the frontend to my backend?**

**Comment by abisar** [comment]
By using commands: https://tauri.app/v1/guides/features/command

**Reply by ambarish999** [reply]
Thanks for replying! I tried that and I could get the inputs from the frontend. But my backend is in the sidecar. How do I pass those inputs to that backend which is python code? Or do I have to write all the backend logic in rust?

**Reply by abisar** [reply]
Your sidecar is a web framework which accepts communication via http requests and not via the command line. You would want to look for running web servers along with tauri which is it's own thing. Take a look at [this question](https://github.com/tauri-apps/tauri/discussions/2751).

This might help:
1. Bundle your webserver (probably with local configs) with your tauri app by using sidecars
2. Run your server upon your app startup
3. Communicate with it via http requests (either from javascript or from rust)

**Reply by ambarish999** [reply]
Not sure I understand completely, but I'll try doing that.
